references/openapi-mock/README.md

```diff
@@ -0,0 +1,215 @@
+
+See below for more details on each feature.
+
+### CORS
+
+The reference mock API proxy makes it easy to test your API from various browser-based clients.
+
+Here's how it works:
+
+* **Automatic CORS Headers:** The proxy automatically adds the necessary CORS headers (like `Access-Control-Allow-Origin`, `Access-Control-Allow-Methods`, etc.) to all responses.
+
+* **Preflight Requests:** The proxy correctly handles preflight `OPTIONS` requests, responding with the appropriate CORS headers to indicate allowed origins, methods, and headers.
+
+* **Permissive Configuration:** By default, the CORS policy is configured to be as permissive as possible, allowing requests from any origin with any HTTP method and headers. This maximizes flexibility for your testing.
+
+The CORS policy ensures that your mock API behaves like a real API in a browser environment, simplifying your development and testing workflow.
+
+### Request Validation
+
+The reference mock API proxy validates the incoming requests against your specification.
+This ensures that the HTTP headers, query parameters, and request body all conform to the defined rules.
+
+This helps you catch errors in your client code early on.
+
+You can disable request validation by passing the header:
+
+```
+Mock-Validate-Request: false
+```
+
+### Dynamic Response Status Code
+
+The reference mock API proxy generates different status codes for your mock API responses. Here's how it works:
+
+* **Prioritizes success:** If the [operation](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.1.1.md#operation-object) allows `HTTP 200` status code, the proxy will use it.
+* **Random selection:** If `HTTP 200` is not allowed for a particular operation, the proxy will pick a random status code from those allowed.
+
+**Want more control?** You can use headers to select response the status code:
+
+* **Specific status code:** Use the `Mock-Status` header in your request and set it to the desired code (e.g., `Mock-Status: 404`).
+* **Random status code:** Use the `Mock-Fuzz: true` header to get a random status code from your spec.
+
+If you use both `Mock-Status` and `Mock-Fuzz`, `Mock-Status` takes precedence.
+
+### Dynamic Response Content-Type
+
+The reference mock API proxy automatically selects the `Content-Type` for responses. Here is how it works:
+
+* **JSON preferred:** If the operation allows `application/json`, the proxy will default to using it.
+* **Random selection:**  If `application/json` is not available, the proxy will randomly choose from the media types available for that operation.
+
+**Want more control?** You can use headers to select the response Content-Type:
+
+* **Standard `Accept` header:** You can use the standard `Accept` header in your request to request a specific media type (e.g., `Accept: application/xml`).
+* **Random media type:**  Alternatively, use the `Mock-Fuzz: true` header to have the proxy select a random media type the available ones.
```

Comment:
Is the same `Mock-Fuzz` header used for random status codes, content-types and response body? How does that work?

Reply:
Yes, `Mock-Fuzz` in general is used a hint to tell the Proxy that you want the dynamic behavior. However, it always has less precedence than the `Mock-Status` and the `Accept` header.